### PR TITLE
Added simple main loop with signal handlers

### PR DESCRIPTION
**Pull request by TheNeikos**
This is for #46.

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -25,10 +25,38 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <signal.h>
+
+static void
+handle_sigint(
+    int sigint
+) {
+    // TODO Do proper logging here
+    printf("Caught signal SIGINT (%d)\n", sigint);
```
Comment (matthiasbeyer) [inline src/main.c]: Can you please change this to write on `stderr` instead?

File: src/main.c [hunk 2]
```diff
@@ -25,10 +25,38 @@
...
+    exit(sigint);
+}
+
+static void
+handle_sigterm(
+    int sigterm
+) {
+    // TODO Do proper logging here
+    printf("Caught signal SIGTERM (%d)\n", sigterm);
```
Comment (matthiasbeyer) [inline src/main.c]: Also here, please on `stderr`.

**Comment by neithernut** [comment]
Er... yeah...
The intention was to have a lib ev loop from the beginning.
Libev also supports a watcher type to listen for signals.

**Comment by TheNeikos** [comment]
Okay! Gonna change to that!

**Comment by neithernut** [comment]
The changes are not that excessive. You can keep the handlers but you have to adjust the definition.
Also: you can use one function for both cases (since you shut down in both cases).
Shutting down can be performed by calling `ev_break` btw.

If you need any information, do not hesitate to ask.

**Comment by TheNeikos** [comment]
For now I used `EVUNLOOP_ALL` with `ev_unloop`. Should I rather use ev_break?

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -56,7 +56,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-unused-parameter -pedantic")
 #
 # With a _real_ libc, we wound't have this problem.
 #
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pthread")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -pthread -lev")
```
Comment (neithernut) [inline src/CMakeLists.txt]: The libev linkage is done somewhere different (but was forgotten). The fix is in PR #42. I'll merge that one, you can then rebase and remove the `-lev`.

**Comment by neithernut** [comment]
`ev_unloop` is acceptable.

File: src/CMakeLists.txt (hunk 3 above)
Comment (TheNeikos) [inline src/CMakeLists.txt]: Oops! That was only a workaround.. shouldn't be in here, you are right.

**Comment by TheNeikos** [comment]
Okay, this is now open for review

**Comment by neithernut** [comment]
As hinted before, the signal handlers may be merged. However, the sollution is acceptable as is.
ACK.

**Comment by TheNeikos** [comment]
I have merged the handlers.

**Comment by neithernut** [comment]
Perfect.
ACK'd, of course.

**Comment by matthiasbeyer** [comment]
Looks good to me so far. ACK.

**Pull request merged**